var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -210,10 +210,20 @@ class Cmake(Package):
     # transparent to patch Spack's versions of CMake's dependencies.
     conflicts("+ownlibs %nvhpc")
 
+    # Use Spack's curl even if +ownlibs, since that allows us to make use of
+    # the conflicts on the curl package for TLS libs like OpenSSL.
+    # In the past we let CMake build a vendored copy of curl, but had to
+    # provide Spack's TLS libs anyways, which is not flexible, and actually
+    # leads to issues where we have to keep track of the vendored curl version
+    # and its conflicts with OpenSSL.
+    depends_on("curl")
```

Comment:
Not sure I get the question

Before this PR: we let CMake build everything it could, but CMake doesn't vendor OpenSSL because of security reasons, so `+ownlibs` never meant CMake was entirely "dependency free", we had to supply a dependency of a vendored dependency.

With this PR: we still let CMake build a bunch of vendored things under `+ownlibs` (compression / archive / json / xml libs), but not curl (and some of curls vendored dependencies).

Reply:
>`+ownlibs` never meant CMake was entirely "dependency free", we had to supply a dependency of a vendored dependency.

Right, which I think we both (and everyone in general) agree that supplying the dep of a vendored dep is... _not good_ to say the least, in that context I agree this change is what we want.

> With this PR: we still let CMake build a bunch of vendored things under `+ownlibs` (compression / archive / json / xml libs), but not curl (and some of curls vendored dependencies).

Yes, this is clear from the PR. I forget CMake has openssl as a bit of a corner case (I just build with native curl and whatever that choses to use as its TLS layer). Disregard my comments.